feat: 🎸 role capabilities

DhariniJeeva opened the pull request:
this pr adds read/delete/update capabilities to roles

randallmorey reviewed (changes requested):
This needs an update test for when authorisation is not present.

DhariniJeeva commented:
> This needs an update test for when authorisation is not present.

added the missing test and removed the jsconfig.files that were added in this [pr](https://github.com/hashicorp/boundary-ui/pull/700)
